datastore/db/bytebase/FertiScan/schema_0.0.9.sql

```diff
@@ -0,0 +1,252 @@
+--Schema creation "fertiscan_0.0.9"
+DO
+$do$
+BEGIN
+IF (EXISTS (SELECT 1 FROM information_schema.schemata WHERE schema_name = 'fertiscan_0.0.9')) THEN
+
+
+    CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+
+    CREATE TABLE "fertiscan_0.0.9"."users" (
+    "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+    "email" text NOT NULL UNIQUE,
+    "registration_date" timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updated_at" timestamp
+    );
+
+    CREATE TABLE "fertiscan_0.0.9"."picture_set" (
+    "id" uuid NOT NULL DEFAULT uuid_generate_v4() PRIMARY KEY,
+    "picture_set" json NOT NULL,
+    "owner_id" uuid NOT NULL REFERENCES "fertiscan_0.0.9".users(id),
+    "upload_date" date NOT NULL DEFAULT current_timestamp,
+    "name" text
+    );
+        
+    alter table "fertiscan_0.0.9".users ADD "default_set_id" uuid REFERENCES "fertiscan_0.0.9".picture_set(id);
+
+    CREATE TABLE "fertiscan_0.0.9"."picture" (
+    "id" uuid NOT NULL DEFAULT uuid_generate_v4() PRIMARY KEY,
+    "picture" json NOT NULL,
+    "nb_obj" int,
+    "picture_set_id" uuid NOT NULL REFERENCES "fertiscan_0.0.9".picture_set(id),
+    "verified" boolean NOT NULL DEFAULT false,
+    "upload_date" TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP
+    );
+
+    CREATE TABLE "fertiscan_0.0.9"."province" (
+    "id" SERIAL PRIMARY KEY,
+    "name" text UNIQUE NOT NULL
+    );
+
+    CREATE TABLE "fertiscan_0.0.9"."region" (
+    "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+    "province_id" int REFERENCES "fertiscan_0.0.9".province(id),
+    "name" text NOT NULL
+    );
+
+    CREATE TABLE "fertiscan_0.0.9"."location" (
+    "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+    "name" text,
+    "address" text NOT NULL,
+    "region_id" uuid References "fertiscan_0.0.9".region(id)
+    );    
+    
+    CREATE TABLE "fertiscan_0.0.9"."organization_information" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "name" text NOT NULL,
+        "website" text,
+        "phone_number" text,
+        "location_id" uuid REFERENCES "fertiscan_0.0.9".location(id)
+    );
+
+    CREATE TABLE "fertiscan_0.0.9"."organization" (
+    "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+    "name" text NOT NULL,
+    "information_id" uuid REFERENCES "fertiscan_0.0.9".organization_information(id),
+    "main_location_id" uuid REFERENCES "fertiscan_0.0.9".location(id)
```

Comment:
Is organization supposed to have `main_location_id` or rather `main_information_id`?

In the latter case, we would end up with such a relation: fertilizer ||--|| organization ||--|| organization_info, with organization bearing no other meaningful information. I would then suggest fertilizer incorporate the `main_information_id`, remove the current organization table and rename organization_info simply organization.

In either case, please consider updating the ERD.

Reply:
I put main_location_id to accelerate the search of it. A company could have a main_location for Communication purposes. I created this field with the thought of having a the company HQ as this ID and any label must have a company location.

Reply:
Still not clear to me, because it seems redundant to associate a location to organization while it is already associated with a organization_information which has a location. But I trust your judgment.